ga4gh/datamodel/variants.py

```diff
@@ -685,15 +687,19 @@ def convertVariant(self, record, callSetIds):
 
         sampleData = record.__str__().split()[9:]  # REMOVAL
         variant.calls = []
-        sampleIterator = 0  # REMOVAL
-        if callSetIds is not None:
-            for name, call in record.samples.iteritems():
-                if self.getCallSetId(name) in callSetIds:
-                    genotypeData = sampleData[sampleIterator].split(
-                        ":")[0]  # REMOVAL
-                    variant.calls.append(self._convertGaCall(
-                        record.id, name, call, genotypeData))  # REPLACE
-                sampleIterator += 1  # REMOVAL
+        if callSetIds is not None and len(callSetIds) > 0:
+            for callSetId in callSetIds:
+                callSet = self.getCallSet(callSetId)
+                if callSet is not None:
+                    call = protocol.Call()
+                    call.callSetId = callSetId
+                    call.callSetName = callSet.getSampleName()
+                    index = self._callSets[callSetId]
+                    genotypeData = sampleData[index].split(
+                             ":")[0]  # REMOVAL
+                    call.genotype, call.phaseset = convertVCFGenotype(
+                             genotypeData, call.phaseset)
+                    variant.calls.append(call)
```

Comment:
I am not a coder on this, so feel free to ignore; however I find it much easier to read code if functions are created rather than nesting loops inside of ifs